From 589d04dc9d5c0daaae8698fbe2d66c088caeef3a Mon Sep 17 00:00:00 2001
From: Roy Zang <tie-fei.zang@freescale.com>
Date: Mon, 21 Jul 2014 13:36:29 -0500
Subject: [PATCH 213/255] layerscape: Fix INTx hang issue

Programming Symbol Timer Register and Filter Mask
Register 1 register bit[31:16] with value 0.

Signed-off-by: Roy Zang <tie-fei.zang@freescale.com>
Signed-off-by: Minghuan Lian <Minghuan.Lian@freescale.com>
[Kevin: Original patch taken from
LS1021A-SDK-V1.1-ARM-SOURCE-20140815-yocto.iso]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/pci/host/pcie-layerscape.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/pci/host/pcie-layerscape.c b/drivers/pci/host/pcie-layerscape.c
index 943c6e4..21c626a 100644
--- a/drivers/pci/host/pcie-layerscape.c
+++ b/drivers/pci/host/pcie-layerscape.c
@@ -34,6 +34,9 @@
 #define SCFG_BIT_REVERSE	0xFFFFFFFF
 #define SCFG_NO_BIT_REVERSE	0x0
 
+/* Symbol Timer Register and Filter Mask Register 1 */
+#define PCIE_STRFMR1 0x71c
+
 /* PCIE ATU configuration registers */
 #define PCIE_ATU_VIEWPORT		0x900
 #define PCIE_ATU_REGION_INBOUND		(0x1 << 31)
@@ -819,6 +822,11 @@ static int ls_pcie_enable(struct ls_pcie *pcie)
 		PCI_COMMAND_MASTER | PCI_COMMAND_SERR;
 	ls_pcie_write_reg(pcie, val, PCI_COMMAND);
 
+	/* Workaround for internal TKT228622 to fix the INTx hang issue */
+	val = ls_pcie_read_reg(pcie, PCIE_STRFMR1);
+	val &= 0xffff;
+	ls_pcie_write_reg(pcie, val, PCIE_STRFMR1);
+
 	dw_pci.nr_controllers = 1;
 	dw_pci.private_data = (void **)&pcie;
 	pci_common_init_dev(pcie->dev, &dw_pci);
-- 
2.0.2

